File: src/gps_subscriber.cpp
#include "ros/ros.h"
#include "first_project/GPS_data.h"

// Topic messages callback
void GPSCallback(const first_project::GPS_data &gps)
{
    ROS_INFO("I received longitude: %.6f", gps.longitude);
    ROS_INFO("I received latitude: %.6f", gps.latitude);
    ROS_INFO("I received bearing: %.6f", gps.heading);
}

int main (int argc, char **argv) {
    // Initiate a new ROS node named "gps_publisher_node"
	ros::init(argc, argv, "gps_subscriber_node");
    //create a node handle: it is reference assigned to a new node
	ros::NodeHandle nh;

	//GPSCallback: is the name of the callback function that will be executed each time a message is received.
    ros::Subscriber sub = nh.subscribe("gps_topic", 1000, GPSCallback);

    // Enter a loop, pumping callbacks
    ros::spin();
}

File: scripts/gps_subscriber.py
#!/usr/bin/env python
import rospy
from first_project.msg import GPS_data

def gps_callback(gps_message):
    rospy.loginfo("new GPS data received: (%.2f, %.2f ,%.2f)", 
        gps_message.longitude,gps_message.latitude,
        gps_message.heading)
    
rospy.init_node('gps_subscriber_node', anonymous=True)

rospy.Subscriber("gps_topic", GPS_data, gps_callback)

# spin() simply keeps python from exiting until this node is stopped
rospy.spin()
